webhook/webhook_test.go

```diff
@@ -0,0 +1,189 @@
+package webhook
+
+import (
+	"testing"
+	"net/http"
+	"net/http/httptest"
+	"log"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"bytes"
+	"io/ioutil"
+
+	"github.com/stretchr/testify/suite"
+	"github.com/gophish/gophish/webhook"
+	"github.com/stretchr/testify/assert"
+)
+
+type WebhookSuite struct {
+	suite.Suite
+}
+
+type mockSender struct {
+	client *http.Client
+}
+
+func newMockSender() *mockSender {
+	ms := &mockSender {
+		client: &http.Client{},
+	}
+	return ms
+}
+
+func (ms mockSender) Send(endPoint webhook.EndPoint, data interface{}) error {
+	log.Println("[test] mocked 'Send' function")
+	return nil
+}
+
+func (s *WebhookSuite) TestSendMocked() {
+	mcSnd := newMockSender()
+	endp1 := webhook.EndPoint{URL: "http://example.com/a1", Secret: "s1"}
+	d1 := map[string]string {
+		"a1": "a11",
+		"a2": "a22",
+		"a3": "a33",
+	}
+	err := mcSnd.Send(endp1, d1)
+	s.Nil(err)
+}
+
+func (s *WebhookSuite) TestSendReal() {
```

Comment:
What are you actually testing here? This is essentially copy/pasting all the code from the signing/sending routines into a test. It's better to call those functions directly.